test/svm/SvmSpoke.Deposit.ts

```diff
@@ -61,17 +92,40 @@ describe("svm_spoke.deposit", () => {
       depositorTokenAccount: depositorTA,
       vault,
       mint: inputToken,
-      tokenProgram: TOKEN_PROGRAM_ID,
+      tokenProgram: tokenProgram ?? TOKEN_PROGRAM_ID,
+      program: program.programId,
     };
   };
 
-  before("Creates token mint and associated token accounts", async () => {
-    await setupInputToken();
-  });
+  const approvedDepositV3 = async (
+    depositDataValues: DepositDataValues,
+    calledDepositAccounts: DepositAccounts = depositAccounts
+  ) => {
+    // Delegate state PDA to pull depositor tokens.
+    const approveIx = await createApproveCheckedInstruction(
```

Comment:
Do we have a test where we catch the error when trying to do this without approving? It might be interesting to document that.

Reply:
sure, I can add that